### ISSUE TITLE
add support for new (2025) actualtext, artifact options of graphicx

### ISSUE THREAD

**Issue by xworld21** [brucemiller/LaTeXML#2679]
graphicx introduced two options at the very end of 2024:
- `actualtext` is additional text for accessibility (this is meant as 'replacement' text, rather than a descriptive text that goes into `alt`)
- `artifact` is a boolean specifying that the image is decorative

Implementation would take only a few lines, but I get the feeling that they will require a schema change and some thinking about the correct HTML equivalent (e.g. if I understand the intention correctly, `actualtext` should probably hide the fact that `<img>` is an image by overriding `@role` and setting `@aria-label` instead of `@alt`).

**Comment by dginev** [comment]
Could you link to the official documentation for these new keys?

**Comment by xworld21** [comment]
The new keys are documented at https://mirrors.ctan.org/macros/latex/required/graphics/grfguide.pdf section 4.4. The documentation does not explain a lot, but the behaviour is meant to match a tagged PDF feature. For ActualText, that seems to be 'replacement', meaning for instance that extracting text from a PDF should literally replace each figure with its ActualText, if present. I keep finding the example of a glyph that is provided as an image but is effectively part of the text.

Some old but relevant sources: ActualText https://www.w3.org/WAI/GL/WCAG-PDF-TECHS-20010913/Overview.html#checkpoint1-4, Artifact https://www.w3.org/TR/WCAG20-TECHS/PDF4.html

**Comment by dginev** [comment]
That's just a spec collision then. I think the right way for us to read `actualtext` is as an alias for `alt`. They are both meant to replace the image when it can not be accessed in its rendered form.

Similarly for `@alt` and `@aria-label`, I think having both is a historical and spec-related accident, but they achieve essentially the same purpose.

**Comment by xworld21** [comment]
> That's just a spec collision then. I think the right way for us to read `actualtext` is as an alias for `alt`. They are both meant to replace the image when it can not be accessed in its rendered form.

Not so much collision, but HTML lacking the distinction. I just checked that `alt` is already used as text replacement if you e.g. copy text with an image and paste it as plain text. I think `actualtext` wants to also inhibit screen readers from spelling "graphic" after an image, but that does not seem possible with HTML without replacing `<img>` with something else entirely (a `<span>` with fixed size and background-image?). That must make decorated drop caps quite annoying.

A simple change would be to map both `actualtext` and `alt` to `alt`, with preference for `alt` which should contain more information. `artifact` should simply enforce `alt=""`, that is, prevent LaTeXML from extracting the description from somewhere else, like a caption.

A more sophisticated solution might be to map `actualtext` to `alt` and `alt` to `aria-description` when both are present.

**Comment by dginev** [comment]
> A simple change would be to map both actualtext and alt to alt, with preference for alt which should contain more information. 

Yes, agreed. Making that more sophisticated may just end up confusing/redundant. 

> artifact should simply enforce alt="", that is, prevent LaTeXML from extracting the description from somewhere else, like a caption.

I think the best practice there is `aria-hidden="true"` instead.
